src/backend/tests/it/upgrade/credentials_init_args.rs

```diff
@@ -0,0 +1,32 @@
+use crate::upgrade::constants::BACKEND_V0_0_25_WASM_PATH;
+use crate::upgrade::types::{ArgV0_0_25, InitArgV0_0_25};
+use crate::utils::mock::CALLER;
+use crate::utils::pocketic::{setup_with_custom_wasm, update_call, upgrade_latest_wasm};
+use candid::{encode_one, Principal};
+use shared::types::token::UserToken;
+use shared::types::user_profile::UserProfile;
+
+#[test]
+fn test_upgrade_credential_init_args() {
+    // Deploy a released canister
+    let arg = ArgV0_0_25::Init(InitArgV0_0_25 {
+        ecdsa_key_name: "master_ecdsa_public_key_fscpm-uiaaa-aaaaa-aaaap-yai".to_string(),
+        allowed_callers: vec![Principal::from_text(CALLER).unwrap()],
+    });
+    let encoded_arg = encode_one(arg).unwrap();
+    let pic_setup = setup_with_custom_wasm(BACKEND_V0_0_25_WASM_PATH, Some(encoded_arg));
+
+    // Test a call
+    let caller = Principal::from_text(CALLER).unwrap();
+    let result = update_call::<Vec<UserToken>>(&pic_setup, caller, "list_user_tokens", ());
+
+    assert!(result.is_ok());
+
+    // Upgrade canister with new wasm
+    upgrade_latest_wasm(&pic_setup)
```

Comment:
LGTM. I was just wondering, given the recent changes added extra field to the `Config`, should we get the `caller_eth_address` before the upgrade, then proceed with the upgrade and get the same address again for comparison purposes? This way, we ensure that even though we added fields in the config, the struct remains intact and the derived key is still derived correctly. Or do you think that would not add much value?

Reply:
That makes a lot of sense.

I had to add the encoded arg as optional parameter to the helper to upgrade as well, but it's still working.

Thanks!